### DISCUSSION TITLE
Image interpolation

### DISCUSSION THREAD

**Discussion by sundeepgoel72** [Bodmer/TFT_eSPI#2643] [General]
Hi,

Is there an **interpolation** function available for resizing (upscale) an image. My use case is  32x24 temperature grid from MLX90640 sensor - which I then display on 320x240 TFT after applying a factor of 7 i.e. each temperature value gets converted into 7x7 rectangle on the display. 

Was wondering if there are any good interpolation libraries which can be applied on the **tft image** or on a **two dimensional array** to get a smooth image based on neighboring values?

regards,

**Comment by Bodmer** [comment]
No, but I did create a sketch for that sensor that upscaled the image using linear interpolation. I will post it in the next few days.

**Reply by sundeepgoel72** [reply]
thank you

**Reply by sundeepgoel72** [reply]
thanks. Have started implementing the code [here ](https://github.com/sundeepgoel72/Thermal-Camera-with-mxl90640-and-ESP32):

**Comment by Bodmer** [comment]
The complete sketch is attached. This is written for a 320x480 display. The upscaled image is 320x240, in addition the raw sensor image is shown and one with a filter applied.
[MLX90640_Interp.zip](https://github.com/Bodmer/TFT_eSPI/files/11501789/MLX90640_Interp.zip)

![ir_cam](https://github.com/Bodmer/TFT_eSPI/assets/15824805/c19cf780-6a07-45cd-a31a-18e958429348)
